### ISSUE TITLE
[REVIEW]: ecopath_matlab: A Matlab-based implementation of the Ecopath food web algorithm

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#64]
**Submitting author:** kakearney (<a href="http://orcid.org/0000-0002-6152-5236">Kelly Kearney</a>)
**Repository:** <a href="https://github.com/kakearney/ecopath_matlab-pkg" target ="_blank">https://github.com/kakearney/ecopath_matlab-pkg</a>
**Version:** v2.0
**Editor:** @Kevin-Mattheus-Moerman 
**Reviewer:** @roliveros-ramos 
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.240977" target="_blank">10.5281/zenodo.240977</a>

## Status

[![status](http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf/status.svg)](http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf"><img src="http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf/status.svg)](http://joss.theoj.org/papers/55cdc396174664e690bec9b2fa7c50bf)
```

**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/kakearney/ecopath_matlab-pkg" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)
### Conflict of interest
- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).
### General checks
- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kakearney/ecopath_matlab-pkg">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v2.0)?
### Functionality
- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?
### Documentation
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
### Software paper

Paper PDF: [10.21105.joss.00064.pdf](https://github.com/openjournals/joss-reviews/files/477566/10.21105.joss.00064.pdf)
- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by arfon** [comment]
/ cc @openjournals/joss-reviewers - would anyone be willing to review this submission?

If you would like to review this submission then please comment on this thread so that others know you're doing a review (so as not to duplicate effort). Something as simple as `:hand: I am reviewing this` will suffice.

**Reviewer instructions**

- Please work through the checklist at the start of this issue.
- If you need any further guidance/clarification take a look at the reviewer guidelines here http://joss.theoj.org/about#reviewer_guidelines
- Please make a publication recommendation at the end of your review

Any questions, please ask for help by commenting on this issue! 🚀

**Comment by rmflight** [comment]
I'm curious if a software package that depends on a *licensed* non-free product counts as **open source software** and is within the scope of the journal?

**Comment by Kevin-Mattheus-Moerman** [comment]
@rmflight yes code requiring a proprietary run-time may be eligible. See also the discussion we had on this here: https://github.com/openjournals/joss/issues/142. Following that discussion we now say the following here http://joss.theoj.org/about: 

> **What about submissions that rely upon proprietary languages/development environments?** 
> We strongly prefer software that doesn't rely upon proprietary (paid for) development environments/programming languages. However, provided your submission meets our submission requirements (including having a valid open source license) then we will consider your submission for review. Should your submission be accepted for review, we may ask you, the submitting author, to help us find reviewers who already have the required development environment installed. 

And at the reviewer guidelines: 

> **What about submissions that rely upon proprietary languages/development environments?**
> As outlined in our author guidelines, submissions that rely upon a proprietary/closed source language or development environment are acceptable provided that they meet the other submission requirements and that you, the reviewer, are able to install the software & verify the functionality of the submission as required by our reviewer guidelines.
> If an open source or free variant of the programming language exists, feel free to encourage the submitting author to consider making their software compatible with the open source/free variant.

**Comment by rmflight** [comment]
I apologize, that was an oversight on my part. I should have looked at the
reviewer guidelines. Thank you for drawing my attention to that.

**Comment by arfon** [comment]
> I apologize, that was an oversight on my part. I should have looked at the reviewer guidelines. Thank you for drawing my attention to that.

👍 no problem.

**Comment by arfon** [comment]
@whedon assign @Kevin-Mattheus-Moerman as editor

**Comment by whedon** [comment]
OK, the editor is @Kevin-Mattheus-Moerman

**Comment by Kevin-Mattheus-Moerman** [comment]
Hi @kakearney Can the authors help to suggest reviewers for this submission?

**Comment by kakearney** [comment]
What is the journal's process for suggesting reviewers?  Should I compile some suggested names and send them to you (the editor), or should I try to personally contact people and direct them to here to volunteer?

**Comment by arfonsmith** [comment]
Please direct them to volunteer here

**Comment by Kevin-Mattheus-Moerman** [comment]
@mdietze @roliveros-ramos @jscamac @tpoisot is this your cup of tea? Would you be willing to act as reviewer for this submission?

**Comment by tpoisot** [comment]
If it runs in Octave, yes.

**Comment by roliveros-ramos** [comment]
It is, I'll be happy to do it.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon assign @roliveros-ramos as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @roliveros-ramos

**Comment by Kevin-Mattheus-Moerman** [comment]
@kakearney I recommend adding a statement in your paper or documentation on compatibility with Octave.

**Comment by Kevin-Mattheus-Moerman** [comment]
@tpoisot Thanks for your quick response.  I've asked the authors to clarify compatibility with Octave. I have just assigned @roliveros-ramos as reviewer 1. For the moment the form at the top here handles a single reviewer (we are expanding our system to easily handle multiple reviewers). If you want to review it as well (if compatible with Octave) that would be great. For you to contribute as a second reviewer you can leave comments here or send me a report via kevin.moerman at gmail.com. Thanks!

**Comment by mdietze** [comment]
Looks like you're covered, and I don't run MATLAB so I wouldn't have been helpful anyway.

**Comment by kakearney** [comment]
I've never worked with Octave myself, so I'll have to download it and run
some tests.  I'll update the paper accordingly once I've had a chance to do
that.

-Kelly

**Comment by Kevin-Mattheus-Moerman** [comment]
@mdietze yes, understood. Thanks for the quick reply!

**Comment by Kevin-Mattheus-Moerman** [comment]
@kakearney It is likely that it is not fully compatible. A basic statement saying that compatibility has not been confirmed could suffice to. But to do a very quick check for compatibility here are some tips. Identify the most complex MATLAB functions in your submission, or functions requiring a special toolbox, or functions that seem very new in MATLAB (I sometimes, somewhat unfairly, refer to Octave as MATLAB 2005). Then you can check if these functions are part of the default Octave distribution. I ask about adding statements on Octave since the open source community in some cases favors non-proprietary languages like Octave.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos thanks again for agreeing to review this submission. Let me know if you have any questions on the review process. Feel free to leave comments to the authors here. Do you think you can process the review within 2 weeks? 

@kakearney any developments on checking compatibility with Octave? Or could you add a statement that is it likely not and has not been checked.

**Comment by kakearney** [comment]
I added a short statement to the README.md file clarifying that this code
will not run in Octave... Octave doesn't support the modern-Matlab class
syntax, along with numerous smaller incompatibilities.

**Comment by roliveros-ramos** [comment]
@Kevin-Mattheus-Moerman Sorry for the late reply, I'm in a meeting and don't have MATLAB in this laptop. I will meet the deadline though.

@kakearney I used MATLAB heavily from 2005-2008, then I moved to R.  I'm very interested in a non-GUI version of EwE, this was really needed. I have MATLAB 2012 installed, do I need to get a more recent version? I think we have 2014.

**Comment by kakearney** [comment]
There are a few key Matlab features that this software relies on that limit its back-compatibility:
- class syntax: introduced in R2008a
- table arrays: introduced in R2013b
- graph/digraph objects: introduced in R2015b

The last of those will only limit the use of the ecopathmodel graph method, which is really a bonus feature, not necessary to the core calculations.  So I believe this code should be mostly functional under R2014a/b.  It will not work properly in R2012.

As I note at the end of the README, this class-based, table-based version of the code is relatively new.  I decided a reliance on newer Matlab features was worth it to create more user-friendly (and input validation-friendly) storage of the hundreds of parameters in an Ecopath model.  In general, I'm  a big proponent of maintaining back-compatibility, and so I did not make the decision to switch to this format lightly; the use of table arrays streamlined both use and maintenance of the code, which had gotten increasingly difficult as I developed this code from a limited, project-specific implementation to a full Ecopath replication useable by others.

Please note that the old version is still available, and the primary ecopath calculation in that code set (ecopathlite.m) uses nearly identical code to the ecopathmodel ecopath method in this one.  That version can be run in Matlab as old as 2007a, I believe (I make ample use of the bsxfun function, which was introduced at that time).  However, I am not actively maintaining that older version anymore.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos Were you able to upgrade MATLAB to review this submission? Let me know if there are any issues.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos are you still able to review this submission? Your help would be greatly appreciated.

**Comment by jscamac** [comment]
Sorry I was away when I was originally tagged in this message, and I forgot to respond.
I don't use Matlab so probably wouldn't be of use.
J

**Comment by roliveros-ramos** [comment]
@Kevin-Mattheus-Moerman, I'm sorry for the lack of news. 
It's been dificult to check the code. I got only MATLAB 2014a for Windows, but I never managed to install the mdbtool library.  I use Fedora but don't have MATLAB for linux available. 
I do have C compilers available but I think the installation process for MS Windows is not as simple as described (not described, actually), mainly for the dependencies of mdbtools itself. After several tries I gave up. I checked the code that doesn't need to use mdbtools and it's working fine. I tried to run the rest of the code with other model included in the demo (Esa) and most of it works properly (despite lots of warnings) so I'd guess it should work fine with the proposed model.  
As a programmer and ecopath user I've dreamed with a tool like this for a lot of time, but my current opinion is an OS restriction should be added because it's not really functional for Windows, which is a pitty since EwE is provided for Windows only and most of the potential users use Windows. Even if for production use may be use in servers running linux (E2E models?), the added value of reproducibility is still appealing for regular EwE users which relay on Windows for their work.
In summary, I cannot fully answer the section of functionality, and in the current state the installation instructions are not available for Windows users. 
Minor stuff: Christensen and Walters 2004 doesn't include a DOI.

**Comment by Kevin-Mattheus-Moerman** [comment]
@jscamac That is okay. Thanks for your response.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos thank you so much for your patience and hard work. If you have a MATLAB license associated with your MathWorks account you can likely download the linux version also. Although the download and installation would require quite a lot of your time.... If the authors are able to provide proper instructions for MS Windows, would you be willing to continue the review process at that point? This would be greatly appreciated. 

@kakearney can you address @roliveros-ramos 's comments? 
Can you consider:
- Fixing the issues for installation under MS Windows
- Adding documentation for installation for MS Windows. 

If your software is not meant for MS Windows users please clarify this in the documentation/instructions and the paper.

**Comment by kakearney** [comment]
Response to reviewer:

> I got only MATLAB 2014a for Windows, but I never managed to install the mdbtool library. I use Fedora but don't have MATLAB for linux available.
I do have C compilers available but I think the installation process for MS Windows is not as simple as described (not described, actually), mainly for the dependencies of mdbtools itself. After several tries I gave up.

The limitation referred to here is related to the mdb2ecopathmodel.m function, one of several tools for importing model data into an ecopathmodel object from other sources. 

To clarify, this limitation does not affect the core capabilities of this code package.  The ecopathmodel class is intended to be the focus, providing both data storage (in class object properties) and a variety of Ecopath-related calculations (in the class methods), and the class can be used completely independently of any external compiled software.

That said, I think the ability for a researcher to move data between the different flavors of Ecopath (the original EwE GUI, Rpath, EwE-F, and my ecopath_matlab) without manual typing or copy-pasting is important, and one of the key features that makes my entry stand out from the crowd.  In addition to being able to create and populate an ecopathmodel object manually from Matlab, users can import from EwE-F text files, Rpath-style .csv files, or EwE6 .EwEmdb files.  The latter one is the troublemaker, and unfortunately also the one most users will want to use the most (since in the Ecopath world, 99% of users build their models using the EwE software). And given that, the lack of cross-platform support by the mdb2ecopath.m code is a very frustrating limitation.

A little history on this limitation and why I haven't figured out a universal solution yet... The original Ecopath with Ecosim software was written using Visual Basic; starting with EwEv6, it was ported to Visual .NET.  For data storage, the authors chose to use Microsoft Access database, i.e. .mdb, files.  This is a proprietary, not-publically-documented file format, and there are no native Matlab functions to read this data.  Therefore, I currently rely on a third-party piece of software, known as mdbtools (https://github.com/brianb/mdbtools), which allows one to dump .mdb tables into simple .csv files, and from there I read them into Matlab.  The mdbtools library is well-documented for *nix systems, but not at all for Windows systems (I assume this is because the target audience is those without access to any Microsoft Access drivers, i.e. non-Windows users).  Based on the discussion threads in the mdbtools GitHub repo, it seems that the library can be compiled on Windows.  However, my Windows use is currently limited to a very small virtual machine, and I have not personally accomplished this Windows compilation.  I am currently investigating workarounds for mdb-reading on Windows using the Matlab Database Toolbox; if offers some utilities to use a locally-installed ODBC driver (available only on Windows) for the task.  I should be able to acquire said ODBC driver for my VM shortly, and hope to rectify this limitation in direct read of .Ewemdb files on a Windows OS as soon as possible.  However, even that solution will be pretty limited, since it requires users to also have the ODBC driver (common for Windows users, I think?) plus a separate Matlab Database Toolbox license (not so common toolbox for oceanographers and fisheries scientists, and certainly not free to add).

I'll note that I state pretty clearly in the tutorial (under "Importing Ecopath with Ecosim data") that the lack of MS Windows support for mdb2ecopathmodel.m is a known shortcoming, one I recognize as affecting many users, and one for which I am actively looking for workarounds.  I also provide copies of both example example ecosystems in both EwE6 format (.mdb files) and Rpath .csv format.  So while Windows users may not be able to work through the specific example that imports directly from the .EwEmdb file, they can still build the same ecopathmodel object using alternative import methods (rpath2ecopathmodel.m).

I am currently working on an additional import function to read EwE6 .csv files.  This set of 13 .csv files can be exported directly from the EwE6 software; it's not ideal, since the files need to be exported manually one at a time (and re-exported if the user modifies the model data at all), but it will serve as a stopgap solution so users can get their data from EwE6 to Matlab without needing the mdbtools library, and without having to resort to copying and pasting.  I will post here where that function is completed an added to the repo.

> I tried to run the rest of the code with other model included in the demo (Esa) and most of it works properly (despite lots of warnings) so I'd guess it should work fine with the proposed model.

Could the reviewer clarify what he meant by "most" of the code working?  Was there a particular part of the tutorial (besides direct import) that didn't work, or that provided different answers than expected?

Regarding the warning messages, I note in the tutorial that those messages are to be expected.  One purpose of this rewrite of Ecopath is to provide as much transparency as possible, so any time my code alters a user-provided value, it issues a warning letting the user know.  This includes any changes related to placeholder values.  In the ESA model example, NaNs were used as placeholders for non-relevant parameters (for example, consumption rate of prey for primary producers that don't eat); the code changes these to 0s (to avoid propagation of NaNs in some later calculations), and the warning messages state clearly where this was done and why.

Each of the above warning messages are coded using standard Matlab warning identifiers, and can be silenced by the user if they so choose.

> In summary, I cannot fully answer the section of functionality, and in the current state the installation instructions are not available for Windows users.

I appreciate the attempt to review, but I feel that this statement judges the entire code package based on a rather minor portion of it.  I will work to correct the direct import on Windows limitation, but I believe the ecopathmodel class can stand on its own even without that feature. If necessary, I can recommend some additional reviewers to take a look at the code.

> Minor stuff: Christensen and Walters 2004 doesn't include a DOI.

The DOI for that paper is 10.1016/j.ecolmodel.2003.09.003.  I have added that information to the paper.bib file.

**Comment by Kevin-Mattheus-Moerman** [comment]
@kakearney thanks for your response. If you can suggest additional reviewers that would be great thanks.

**Comment by kakearney** [comment]
I don't want to post emails to this thread, so I've emailed a couple
potential reviewers, and directed them to volunteer in this thread if they
are willing.

**Comment by roliveros-ramos** [comment]
@kakearney: I do understand mdbtools is required for the mdb2ecopathmodel function only and I don't know if I miss the purpose of the package or not, but your script showing the functionality of the package relies in the Gen37 and Tb models, and both of them requires the `mdb2ecopathmodel` function, so I doesn't matter. I do know it is possible to install mdbtools on MS Windows but sadly I couldn't do it and I couldn't find instructions on the internet helpful to me. You also have to consider the dependencies for mdbtools need to be installed too:

> Installation: Does installation proceed as outlined in the documentation?
> Installation instructions: Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.

I cannot check those points. If the process is simple it will be of great value to have add them to the installation instructions since, as you said, most EwE users work with the mdb format and MS Windows, and most EwE users are not particularly computer-savvy. That said, with MS Windows and MATLAB 2014a I was not able to run the code, reason why I cannot fully check the items in the review I mentioned.

Despite that, I run the script doing `Gen37=Esa` and `Tb=Esa` with minor changes in the names of the species (e.g. for the bootstrap). The code run but I got errors with most of the plots. Most of them can be related to the 2014a version I'm using (missing histogram function and methods for 'digraph') but I'm not sure (`Error using table/subsrefDot (line 117) Index exceeds matrix dimensions.`), but I do understand the full code was not intended to be run with the Esa model.  A quick fix can be including Gen37 and Tb as csv files so Windows users can test the core functionality of the package without the mdbtools dependency, but I would stress that it reduces the usability of the package for Windows users (myself have several EwE models in the regular mdb format I cannot inmediately try with your package in the current stage). So, please, take my review as the point of view of a EwE and Windows user, I just learned with your article about Rpath, for example.   

@Kevin-Mattheus-Moerman: If the authors provide installation instructions for MS Windows or find an alternative solution for MS Windows users I'd like to continue with the review, I'd be able to borrow a MATLAB 2016a.

**Comment by kakearney** [comment]
Thank you for the additional clarification.

As I was preparing to respond to this comment, I received an email from a new ecopath_matlab user asking if I had a workaround for the mdbtools limitation on Windows.  So while I may consider that function a bonus feature, it's now obvious to me that users need a solution to import existing EwE6 models directly on a Windows system.  I will try to get the EwE6 .csv import option working as soon as possible, and will add this option to the tutorial.  I think this should take me about a week.

> The code run but I got errors with most of the plots. Most of them can be related to the 2014a version I'm using (missing histogram function and methods for 'digraph') but I'm not sure

The `histogram` function was introduced in Matlab R2014b (intended to replace `hist` and `histc`), and graph/digraph objects in R2015b.  The histogram plot was simply demonstrating distribution of parameters; I'll go ahead and change that code to use a more back-compatible option. 

> but I do understand the full code was not intended to be run with the Esa model.

While the exact verbatim code in the tutorial was designed around several different models (Gen37 represents the simplest sort, Tampa Bay a more complex one with multi-stanza groups and multiple fishing fleets, and Esa a manually-built one with the sort of parameters often included in print journals), all the ecopathmodel methods should work with any set of model data.  

> A quick fix can be including Gen37 and Tb as csv files so Windows users can test the core functionality of the package without the mdbtools dependency, but I would stress that it reduces the usability of the package for Windows users (myself have several EwE models in the regular mdb format I cannot inmediately try with your package in the current stage)

The models are currently included in the examples folder in Rpath csv format.  I'll add EwE6 csv format shortly (as well as instructions for exporting models from EwE6 into this format).

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos please can you pick up the review process again and consider the comments above by @kakearney

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos best wishes for the new year. Are you able to pick up the review process? Thanks.

**Comment by roliveros-ramos** [comment]
@Kevin-Mattheus-Moerman, I have finished the review. I really appreciate the efforts from @kakearney  to make it easier to use to Windows users.  The python functions works from MATLAB 2014b, so I changed from 2014a to 2016a, which is the MATLAB version I used to this review. I was able to fully run the demo included with the package and it's working as expected, I still have some comments:

1. `mdb2ecopathmodel` function: I needed to change the `mdbexport.py` (line 33) to:
`DRV = '{Microsoft Access Driver (*.mdb, *.accdb)}'`
because the name of the driver in my computer was not the one provided by default. Without the proper driver name, you get the error:

> python error: [im002] [microsoft][odbc driver manager] datasource name not found and no default driver specified

 I'd be nice to have a warning or some workaround for that.

2. `ecopathmodel2rpath` function. It works ok, but cannot be reverted using the `rpath2ecopathmodel` functions:
`Gen37 = rpath2ecopathmodel(fullfile('examples', 'Gen37', 'gen37'));`
 since more files are required than the ones written by the function (e.g. `Unable to open file 'examples\Gen37\gen37_pedigree.csv'.`). I made it work creating empty "pedigree", "stanzas" and "stanza_groups" (just with the proper headers). The same applied to the TampaBay model.

3. The headers of the file `tb_stanza_groups.csv` are incorrect, `VGFB` need to be changed to `VBGF` to make it work.  

One additional comment which may be useful for the installation instructions: MS Office installed by default is the 32 bits one if you already have other 32bits MS Office stuff (any driver or viewer, very likely if you have your computer more than a few years for example), so normally ODBC drivers are 32 bits (or it was my case). If your MATLAB version is 64 bits you need to install Python 64 bits and have 64 bits ODBC drivers. You may have to reinstall some of the programs to make them compatible. I reinstalled MS Office 64bits since we have licenses for that. Not sure if it's the only solution, but was the simpler.

**Comment by Kevin-Mattheus-Moerman** [comment]
Great work! Thanks a lot @roliveros-ramos ! @kakearney can you please address these additional comments?

**Comment by kakearney** [comment]
Thank you for the updated comments.  I've addressed them as follows:

1.  I added a few additional lines of code to the mdbexport.py function to list all available drivers and choose one that included 'Microsoft Access Driver' in its name.  That will at the least address the two versions of the driver name that I now know of.  I'm not sure if that covers all possible options, but I will try to update this in the future if I discover other variations on the driver name.

2.  I changed the syntax of the rpath2ecopathmodel function to now match that of the read.rpath.params R function that it was designed to sort-of mimic.  Under this syntax, only the base model table and diet table files are required; all others are optional, and assumed to be irrelevant if not explicitly included by the user.  This unfortunately breaks back-compatibility with my old code, which I had been trying to avoid, but I think the new similarity in syntax between the R code and my Matlab code will make things much simpler for users trying to move between the two platforms.  I updated the ecopathmodel_overview.m file to reflect this new syntax.

3.  Tracked this to a typo in the ecopathmodel2rpath.m function.  I corrected it both there and in Tampa Bay .csv file.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos can you review how @kakearney addressed your comments? Let me know if you have additional comments or if you feel this submission is now suitable for publication.

**Comment by roliveros-ramos** [comment]
I just pulled the lasted code and I'm getting this error:

> Error using mdb2ecopathmodel>checkdeps (line 531)
> Could not import mdbexport python module; please check that you either installed it locally or kept it in its default location in the ecopath_matlab package
> 
> Error in mdb2ecopathmodel (line 106)
> checkdeps;

Points 2 and 3 are fine!

**Comment by kakearney** [comment]
Sorry, pushed that previous update without testing it; a small syntax error (spaces instead of tabs) prevented Matlab from properly importing the module.  Just corrected that (and tested it on Windows this time)... should be good to go now.

**Comment by Kevin-Mattheus-Moerman** [comment]
@kakearney great, looks like we are getting close. @roliveros-ramos can you please confirm?

**Comment by roliveros-ramos** [comment]
@Kevin-Mattheus-Moerman: everything looks fine now, I recommend the paper is accepted for publication.

**Comment by Kevin-Mattheus-Moerman** [comment]
@roliveros-ramos great. Thanks for your awesome role as reviewer! 
@kakearney congratulations! 🍰 We will process acceptance of your work shortly.

**Comment by arfon** [comment]
@kakearney - At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by arfon** [comment]
BTW, here's the latest paper: 
[10.21105.joss.00064.pdf](https://github.com/openjournals/joss-reviews/files/702296/10.21105.joss.00064.pdf)

**Comment by kakearney** [comment]
I just issued a new release, v2.1.  The new DOI is 10.5281/zenodo.17838

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.240977 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.240977" target="_blank">10.5281/zenodo.240977</a> is the archive.

**Comment by arfon** [comment]
@roliveros-ramos many thanks for your review here and to you for editing @Kevin-Mattheus-Moerman ✨ 

@kakearney - your paper is now accepted into JOSS and your DOI is http://joss.theoj.org/papers/10.21105/joss.00064 ⚡️ 🚀 💥
